time t
float32[] x_p
float32 impulse
